dipy/viz/ui.py

```diff
@@ -2618,3 +2619,186 @@ def set_center(self, position):
+        state change.
+
+        Parameters
+        ----------
+        imgPath : string
+            Path of the image
+
+        Returns
+        -------
+        img : vtkImageDataGeometryFilters
+            The corresponding image .
+
+        """
+        if imgPath.split(".")[-1] in ["png", "PNG"]:
+            png = vtk.vtkPNGReader()
+            png.SetFileName(imgPath)
+            png.Update()
+            img = png.GetOutput()
+        elif imgPath.split(".")[-1] in ["jpg", "jpeg", "JPG", "JPEG"]:
+            jpeg = vtk.vtkJPEGReader()
+            jpeg.SetFileName(imgPath)
+            jpeg.Update()
+            img = jpeg.GetOutput()
+        else:
+            error_msg = "This file format is not supported by the Image Holder"
+            warn(Warning(error_msg))
+        return img
+
+    @property
+    def size(self):
+        """ Gets the image size.
+
+        """
+        return self._size
+
+    @size.setter
+    def size(self, size):
+        """ Sets the image size.
+
+        Parameters
+        ----------
+        size : (float, float)
+            Image size (width, height) in pixels.
+
+        """
+        self._size = np.asarray(size)
+
+        # Update actor.
+        self.texture_points.SetPoint(0, 0, 0, 0.0)
+        self.texture_points.SetPoint(1, size[0], 0, 0.0)
+        self.texture_points.SetPoint(2, size[0], size[1], 0.0)
+        self.texture_points.SetPoint(3, 0, size[1], 0.0)
+        self.texture_polydata.SetPoints(self.texture_points)
+
+    def scale(self, size):
+        """ Scales the image.
+
+        Parameters
+        ----------
+        size : (float, float)
+            Scaling factor (width, height) in pixels.
+
+        """
+        self.size *= size
+
+    def build_actor(self):
+        """ Return an image as a 2D actor with a specific position.
+
+        Returns
+        -------
+        :class:`vtkTexturedActor2D`
+
+        """
+        img = self.img
+        self.texture_polydata = vtk.vtkPolyData()
+        self.texture_points = vtk.vtkPoints()
+        self.texture_points.SetNumberOfPoints(4)
+        self.size = img.GetExtent()
+
+        polys = vtk.vtkCellArray()
+        polys.InsertNextCell(4)
+        polys.InsertCellPoint(0)
+        polys.InsertCellPoint(1)
+        polys.InsertCellPoint(2)
+        polys.InsertCellPoint(3)
+        self.texture_polydata.SetPolys(polys)
+
+        tc = vtk.vtkFloatArray()
+        tc.SetNumberOfComponents(2)
+        tc.SetNumberOfTuples(4)
+        tc.InsertComponent(0, 0, 0.0)
+        tc.InsertComponent(0, 1, 0.0)
+        tc.InsertComponent(1, 0, 1.0)
+        tc.InsertComponent(1, 1, 0.0)
+        tc.InsertComponent(2, 0, 1.0)
+        tc.InsertComponent(2, 1, 1.0)
+        tc.InsertComponent(3, 0, 0.0)
+        tc.InsertComponent(3, 1, 1.0)
+        self.texture_polydata.GetPointData().SetTCoords(tc)
+
+        texture_mapper = vtk.vtkPolyDataMapper2D()
+        if major_version <= 5:
```

Comment:
I know it's not your fault, but `major_version` is a terrible name. The major version of what? @skoudoro do you have an opinion on this? Should we rename this? And maybe create a function with the patterns

    if version SetInput else SetInputData
because it's used 4 times only in this file. We might also wonder if there's these still some people using vtk5. It's been 5 years since vtk6.

Reply:
+1 @karandeepSJ please look into this.

Reply:
This looks like a good use for the viz utility function [set_input](https://github.com/nipy/dipy/blob/4484514fb039bc212bc4e503829195c02e35ab95/dipy/viz/utils.py#L17)

Reply:
I'll create a separate PR for this.